### PR TITLE
[integration test] confirm branch

### PR DESCRIPTION
**Pull request by acha-bill**
# Description of change

Try to confirm a branch by issuing enough messages to build AW for the branch.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tools/integration-tests/tester/tests/testutil.go [hunk 1]
```diff
@@ -619,3 +620,131 @@ func txMetadataStateEqual(t *testing.T, node *framework.Node, txID string, expIn
...
+	if err != nil {
+		return
+	}
+	attachments, err := peer.GetTransactionAttachments(branchID)
+	if err != nil {
+		return
+	}
+	children := attachments.MessageIDs
+	seen := make(map[string]struct{})
+	for len(children) > 0 {
```
Comment (karimodm) [inline tools/integration-tests/tester/tests/testutil.go:649]: This loop over `children`, preventing to process a msgID twice is kinda hard to read. Hive.go has a datastructure exactly for cases like these, check it out https://github.com/iotaledger/hive.go/blob/5c76c0ccc6682e898bd45681ad5f3502b5f3ee50/datastructure/walker/walker.go

Could make the code significantly cleaner!

File: tools/integration-tests/tester/tests/testutil.go [hunk 2]
```diff
@@ -619,3 +620,131 @@ func txMetadataStateEqual(t *testing.T, node *framework.Node, txID string, expIn
...
+		}
+		seen[msgID] = struct{}{}
+		tip = msgID
+		var msg *jsonmodels.Message
+		msg, err = peer.GetMessage(msgID)
+		if err != nil {
+			return
+		}
+		for _, approver := range msg.StrongApprovers {
+			if msg, err := peer.GetMessage(approver); err == nil {
```
Comment (karimodm) [inline tools/integration-tests/tester/tests/testutil.go:663]: Can you rename this `msg` so it doesn't shadow the outer `msg`?

**Review by karimodm — COMMENTED** [review]
Thanks for the PR! Just some comments since it is still a Draft.

File: packages/jsonmodels/tangle.go [hunk 3]
```diff
@@ -8,18 +8,20 @@ import (
...
-	SequenceNumber  uint64   `json:"sequenceNumber"`
-	PayloadType     string   `json:"payloadType"`
-	TransactionID   string   `json:"transactionID,omitempty"`
-	Payload         []byte   `json:"payload"`
-	Signature       string   `json:"signature"`
+	ID                    string   `json:"id"`
+	StrongParents         []string `json:"strongParents"`
+	WeakParents           []string `json:"weakParents"`
+	ShallowLikeParents    []string `json:"shallowLikeParents"`
+	ShallowDislikeParents []string `json:"shallowDislikeParents"`
```
[the remark this answers is not in the document]
Comment (acha-bill) [reply packages/jsonmodels/tangle.go:15]: We'll see when it's time to merge. :)

**Review by jkrvivian — APPROVED** [review]

File: tools/integration-tests/tester/tests/testutil.go [hunk 4]
```diff
@@ -673,3 +676,130 @@ func TryConfirmMessage(t *testing.T, n *framework.Network, requiredPeers []*fram
...
+						continue
+					}
+
+					approverLikes := false
+					for _, like := range approverMsg.ShallowLikeParents {
+						if like == branch.ID {
+							approverLikes = true
+							break
+						}
+					}
```
Comment (daria305) [inline tools/integration-tests/tester/tests/testutil.go:730]: I don't get this one, we compare here messageID and branchID

**Review by daria305 — COMMENTED** [review]
I found one strange thing, besides that all alright

**Pull request merged**